### PR TITLE
Doro67 cleanup: Add spacing to LAYOUT macros, add layout comments, improve consistency, fix ISO layout bug

### PR DESCRIPTION
**Pull request by vomindoraan**
## Description

Doro67 has three available PCBs: `multi`, `regular` and `rgb`. This PR:
- cleans up the default keymaps for all three PCBs;
- makes them more consistent with each other;
- adds spacing and alignment to `LAYOUT` macros for improved visibility;
- adds pretty layout comments feat. box drawing characters to keymaps and readme files;
- reformats the code to conform to current QMK code standards (mostly 4-space indentation);
- fixes a bug with the positions of Enter and NUHS keys in `LAYOUT_iso` for the `multi` PCB.

Support for Doro67 was initially added in #5115, #5539, #5677. Tagging @mechmerlin and @ShadeDream, the authors of those PRs, for review.

## Types of Changes

- [ ] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by vomindoraan** [comment]
I still find the choice of `KC_F1` and `KC_F2` for the extra keys (split LShift and split Backspace) on the `multi` PCB very odd. Should I change them to `KC_NUBS` and `KC_GRV`, respectively, and make the split Backspace actually like a split Backspace (i.e. `KC_BSPC` instead of `KC_BSLS`)?

File: keyboards/doro67/multi/keymaps/default_multi/keymap.c [hunk 1]
```diff
@@ -15,22 +15,46 @@
...
+     * │LSft│F1 │ Z │ X │ C │ V │ B │ N │ M │ , │ . │ / │RShift│ ↑ │PgD│
+     * ├────┼───┴┬──┴─┬─┴───┴───┴┬──┴─┬─┴───┴──┬┴──┬┴──┬┴──┬───┼───┼───┤
+     * │LCtl│LGui│LAlt│  Space   │Spc │ Space  │RAl│Fn │RCt│ ← │ ↓ │ → │
+     * └────┴────┴────┴──────────┴────┴────────┴───┴───┴───┴───┴───┴───┘
+     */
+    [0] = LAYOUT_multi(
+        KC_ESC,  KC_1,    KC_2,    KC_3,    KC_4,    KC_5,    KC_6,    KC_7,    KC_8,    KC_9,    KC_0,    KC_MINS, KC_EQL,  KC_BSPC, KC_F2,   KC_INS,
+        KC_TAB,  KC_Q,    KC_W,    KC_E,    KC_R,    KC_T,    KC_Y,    KC_U,    KC_I,    KC_O,    KC_P,    KC_LBRC, KC_RBRC, KC_BSLS,          KC_DEL,
+        KC_CAPS, KC_A,    KC_S,    KC_D,    KC_F,    KC_G,    KC_H,    KC_J,    KC_K,    KC_L,    KC_SCLN, KC_QUOT,          KC_ENT,           KC_PGUP,
+        KC_LSFT, KC_F1,   KC_Z,    KC_X,    KC_C,    KC_V,    KC_B,    KC_N,    KC_M,    KC_COMM, KC_DOT,  KC_SLSH, KC_RSFT,          KC_UP,   KC_PGDN,
```
Comment (vomindoraan) [inline keyboards/doro67/multi/keymaps/default_multi/keymap.c]: I still find the choice of `KC_F1` and `KC_F2` for the extra keys (split LShift and split Backspace) on the `multi` PCB very odd. Should I change them to `KC_NUBS` and `KC_GRV`, respectively, and make the split Backspace actually like a split Backspace (i.e. swap `KC_BSPC` and `KC_BSLS`)?

Comment (vomindoraan) [reply keyboards/doro67/multi/keymaps/default_multi/keymap.c]: I decided to do that.

**Comment by vomindoraan** [comment]
During testing on my `multi` PCB, I noticed that the original `LAYOUT_iso` had a bug where the positions of Enter and NUHS were flipped in the physical section of the macro. I just pushed a commit that fixes that, so this is now a bugfix PR as well.

The other `multi` layouts are tested and working correctly.

File: keyboards/doro67/rgb/keymaps/default/keymap.c [hunk 2]
```diff
@@ -17,46 +17,75 @@
...
-        // when keycode QMKBEST is released
-      }
-      break;
+        if (record->event.pressed) {
+            // When keycode QMKBEST is pressed
+            SEND_STRING("QMK is the best thing ever!");
+        } else {
+            // When keycode QMKBEST is released
+        }
+        return false;
```
Comment (drashna) [inline keyboards/doro67/rgb/keymaps/default/keymap.c]: Actually, could you revert these changes, please? 

There are some unexpected behaviors if you return false.   Namely, it doesn't handle one shot keys correctly, and some other stuff. 

In general, you should ONLY return false, if you are replacing an existing key's behavior.

Comment (vomindoraan) [reply keyboards/doro67/rgb/keymaps/default/keymap.c]: Why does this break one shot keys if these are custom keycodes?

Comment (drashna) [reply keyboards/doro67/rgb/keymaps/default/keymap.c]: Because the handling happens late in the stack.  returning false means that it never gets to that point, and never clears the one shot status.

Comment (vomindoraan) [reply keyboards/doro67/rgb/keymaps/default/keymap.c]: Is this a new thing or has it always been like this? Because I've always done custom keycodes like this, and seen it done this way in lots of places, and I've never had any problems with one shot keys.

Comment (drashna) [reply keyboards/doro67/rgb/keymaps/default/keymap.c]: It's always been that way.  It's just doesn't come up too often.  

And unless there is a specific reason to return false, it is better to return true.

**Comment by vomindoraan** [comment]
@noroadsleft @drashna I believe I've resolved the issues.

**Comment by vomindoraan** [comment]
Rebased onto master to fix the LUFA submodule issue.
